tools/secure_notebook.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python
+
+from notebook.auth import passwd
+from traitlets.config.loader import JSONFileConfigLoader, ConfigFileNotFound
+import six
+from jupyter_core.paths import jupyter_config_dir
+from traitlets.config import Config
+
+
+
+from OpenSSL import crypto, SSL
+from socket import gethostname
+from pprint import pprint
+from time import gmtime, mktime
+from os.path import exists, join
+
+import io
+import os
+import json
+
+
+def create_self_signed_cert(cert_dir, keyfile, certfiile):
+    """
```

Comment:
Let's get into the habit of always honoring the docstring conventions for parameter descriptions... 

Also, typo in `certfiile`.